### PR TITLE
add support for composite builds

### PR DESCRIPTION
**Pull request by tschellenbach**
support for composite builds

- android gradle to 4.0.1
- gradle to 6.6.1
- remove outdated jacoco-android package
- rename library -> client
- use github actions in stead of travis

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - name: set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: KtLint
+        run: bash ./gradlew ktlintCheck
+      - name: Unit tests
+        run: bash ./gradlew test --stacktrace
+      - name: Coverage
+        run: ./gradlew jacocoTestReport
```
Comment (zsmb13) [inline .github/workflows/android.yml:27]: I believe we don't need `bash` in front of any of these

File: sample/build.gradle [hunk 2]
```diff
@@ -4,6 +4,8 @@ apply plugin: 'kotlin-android-extensions'
 apply plugin: 'kotlin-kapt'
 apply plugin: 'com.google.gms.google-services'
 apply plugin: 'org.jlleitschuh.gradle.ktlint'
+apply plugin: "com.hiya.jacoco-android"
```
Comment (JcMinarro) [inline sample/build.gradle]: Why do we need jacoco into our sample app?

Comment (tschellenbach) [reply sample/build.gradle]: we dont

**Comment by tschellenbach** [comment]
merging with master is giving issues. will restart from master branch and open a clean PR

**Pull request closed without merging**